include/xtensor/xmath.hpp

```diff
@@ -2610,8 +2610,8 @@ namespace detail {
               XTL_REQUIRES(is_reducer_options<EVS>)>
     inline auto nanvar(E&& e, EVS es = EVS())
     {
-        decltype(auto) sc = detail::shared_forward<E>(e);
-        return nanmean<T>(square(sc - nanmean<T>(sc)), es);
+        auto cached_mean = nanmean<T>(e, es)();
+        return nanmean<T>(square(std::forward<E>(e) - cached_mean), es);
```

Comment:
I don't understand why a `move` is required in the corresponding implementation of `variance`
https://github.com/xtensor-stack/xtensor/blob/54dbb8223d5a874261e8e988b487054d2e50710e/include/xtensor/xmath.hpp#L2052
. Here `cached_mean` is a scalar right?

Reply:
Yes, I the `std::move` an oversight from a previous refactoring.

Reply:
Thanks!

Reply:
Actually I think this move is still required, even if `cached_mean` is a scalar. Scalars are not a special case, they are captured the same way as expressions, so we must move them to avoid dangling references.

Reply:
I see. Thank you for the explanation!